pandas/tests/extension/base/reduce.py

```diff
@@ -28,15 +69,19 @@ def test_reduce_series_numeric(self, data, all_numeric_reductions, skipna):
         op_name = all_numeric_reductions
         s = pd.Series(data)
 
-        with pytest.raises(TypeError):
+        msg = generate_error_message(s.dtype.name, op_name)
+
+        with pytest.raises(TypeError, match=msg):
```

Comment:
rather than adding a new function, maybe we could just check
```python
msg = "[Cc]annot perform|Categorical is not ordered for operation|'Categorical' does not implement reduction"
```